k2/csrc/fsa_utils.cu

```diff
@@ -1026,7 +1028,8 @@ Array1<FloatType> GetForwardScores(FsaVec &fsas, Ragged<int32_t> &state_batches,
           num_arcs = fsas.TotSize(2);
   int32_t num_batches = state_batches.Dim0();
   // just using DCHECK below to save time in production code
-  K2_DCHECK_EQ((state_batches.TotSize(1) / num_batches), num_fsas);
+  K2_DCHECK(num_batches == 0 ||
```

Comment:
We chould just do: `K2_DCHECK_EQ(state_batches.TotSize(1), num_batches * num_fsas);`...

Reply:
yes, multiplication is faster than division.

Reply:
sure, will update